public/scripts/world-info.js

```diff
@@ -2123,11 +2161,15 @@ async function displayWorldEntries(name, data, navigation = navigation_option.no
         // Convert the data.entries object into an array
         let entriesArray = Object.keys(data.entries).map(uid => {
             const entry = data.entries[uid];
+            if (!entry || typeof entry !== 'object') {
+                return null;
+            }
             entry.displayIndex = entry.displayIndex ?? entry.uid;
             return entry;
-        });
+        }).filter(entry => entry !== null);
 
         // Apply the filter and do the chosen sorting
+        entriesArray = addMissingWorldInfoFields(entriesArray);
```

Comment:
Shouldn't this be called on loading a WI from the server side?
Like in `loadWorldInfo`, or something similar.

I know the original bug report said display was bugged and scanning worked, but having missing fields or invalid fields *could* also lead to unexpected results in processing WI for matches.